ttyplot.c

```diff
@@ -277,40 +274,12 @@ static void paint_plot(void) {
     move(0,0);
 }
 
-static void resize(int signum) {
-    (void)signum;
-    sigwinch_pending = 1;
-}
-
-static void finish(int signum) {
-    (void)signum;
-    sigint_pending = 1;
-}
-
-static int pselect_without_signal_starvation(
-        int nfds,
-        fd_set * readfds,
-        fd_set * writefds,
-        fd_set * exceptfds,
-        const struct timespec * timeout,
-        const sigset_t * sigmask) {
-    // With high pressure on file descriptors (e.g. with "ttyplot < /dev/zero")
-    // a call to `pselect` could stall signal delivery for 20+ seconds on Linux.
-    // To avoid that situation, we first do a call to `pselect` that is dedicated
-    // to signal delivery and that only.
-    // (Related: https://stackoverflow.com/q/62315082)
-    const struct timespec zero_timeout = { .tv_sec = 0, .tv_nsec = 0 };
-
-    // First call, signal delivery only
-    const int select_ret = pselect(0, NULL, NULL, NULL, &zero_timeout, sigmask);
-
-    const bool signal_received = ((select_ret == -1) && (errno == EINTR));
-    if (signal_received) {
-        return select_ret;
-    }
-
-    // Second call
-    return pselect(nfds, readfds, writefds, exceptfds, timeout, sigmask);
+// Send signals through a pipe, in order to catch them without race conditions.
+// pselect() could be an alternative, but it is unreliable on Linux.
+// (Related: https://stackoverflow.com/q/62315082)
+static void signal_handler(int signum) {
+    unsigned char signal_number = signum;
```

Comment:
@edgar-bonet this downcast is probably only safe because we know we only receive SIGWINCH and SIGINT here? I think it could use a comment, maybe even also a cast, I think MSVC would complain here.

Reply:
More generally, signal numbers are all small integers. I added a cast and a comment.